### PR TITLE
i18n OSD Elements

### PR DESCRIPTION
**Pull request by McGiverGim**
This is a big PR to let translate all the OSD elements. I have reordered, added missing elements, etc. to the messages file.

The translators will have A LOT of work, so let's try to do it right the first time. Two things:

- Now that we order the elements by text, maybe is a good moment to decide if we want to modify some of them. For example, modify `Main battery voltage` and others battery related elements to start all of them with `Battery`, in this way they will appear all of them next to the other in the list. The same for example with `GPS`. I'm not a native English, so I need help from others here because my phrases are in a very standard form and I can't be sure what is valid or not here. With a review will be enough but a PR to this branch will be maybe better if they are a lot.
- Maybe modify some of them to give a more detailed text (we have the description too so it is not a description). I have modified some of them (`avg` for `average` for example).

@mikeller what do you think? We let the texts as is now or we modify them before uploading to the translators?

### CODE REVIEW

File: locales/en/messages.json [hunk 1]
```diff
@@ -3766,186 +3766,454 @@
         "message": "Reset Settings"
     },
 
+    "osdTextElementMainBattVoltage": {
+        "message": "Main battery voltage",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery voltage'

File: locales/en/messages.json [hunk 2]
```diff
@@ -3766,186 +3766,454 @@
...
+    },
+    "osdTextElementVtxChannel": {
+        "message": "VTX channel",
+        "description": "One of the elements of the OSD"
+    },
+    "osdDescElementVtxChannel": {
+        "message": "Current VTX channel and power"
+    },
+    "osdTextElementVoltageWarning": {
+        "message": "Voltage warning",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery voltage warning'

File: locales/en/messages.json [hunk 3]
```diff
@@ -3766,186 +3766,454 @@
...
     },
+    "osdTextElementArtificialHorizon": {
+        "message": "Artificial horizon",
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementArtificialHorizon": {
         "message": "Graphical artificial horizon indicator"
     },
+    "osdTextElementHorizonSidebars": {
+        "message": "Horizon sidebars",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Artificial horizon side bars'

File: locales/en/messages.json [hunk 4]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementHorizonSidebars": {
         "message": "Sidebars around artificial horizon indicator"
     },
+    "osdTextElementCurrentDraw": {
+        "message": "Current draw",
+        "description": "One of the elements of the OSD"
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery current draw'

File: locales/en/messages.json [hunk 5]
```diff
@@ -3766,186 +3766,454 @@
...
+    },
     "osdDescElementCurrentDraw": {
         "message": "Instantaneous battery current draw"
     },
+    "osdTextElementMahDrawn": {
+        "message": "MAh drawn",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery current MAh drawn'

File: locales/en/messages.json [hunk 6]
```diff
@@ -3766,186 +3766,454 @@
...
-        "message": "Display name as set by the \"display_name\" cli command"
+    "osdTextElementAltitude": {
+        "message": "Altitude",
+        "description": "One of the elements of the OSD"
     },
     "osdDescElementAltitude": {
         "message": "Current altitude (flashes when above alarm threshold)"
     },
+    "osdTextElementOnTime": {
+        "message": "On time",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Timer: On time'

File: locales/en/messages.json [hunk 7]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementOnTime": {
         "message": "Total time the craft has been powered on"
     },
+    "osdTextElementFlyTime": {
+        "message": "Fly time",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Timer: Flight time'

File: locales/en/messages.json [hunk 8]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementFlyTime": {
         "message": "Total time the craft has been armed in the current power cycle (flashes when above alarm threshold)"
     },
+    "osdTextElementFlyMode": {
+        "message": "Fly mode",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Flight mode'

File: locales/en/messages.json [hunk 9]
```diff
@@ -3766,186 +3766,454 @@
...
     },
     "osdDescElementDebug": {
         "message": "Debug variables"
     },
+    "osdTextElementPIDRoll": {
+        "message": "PID roll",
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementPIDRoll": {
         "message": "Roll axis PID gains"
```
Comment (mikeller) [inline locales/en/messages.json]: 'PID: Roll axis gain'

File: locales/en/messages.json [hunk 10]
```diff
@@ -3766,186 +3766,454 @@
...
+    },
+    "osdTextElementDebug": {
+        "message": "Debug",
+        "description": "One of the elements of the OSD"
     },
     "osdDescElementDebug": {
         "message": "Debug variables"
     },
+    "osdTextElementPIDRoll": {
+        "message": "PID roll",
```
Comment (mikeller) [inline locales/en/messages.json]: 'PID: Roll'

File: locales/en/messages.json [hunk 11]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementPIDRoll": {
         "message": "Roll axis PID gains"
     },
+    "osdTextElementPIDPitch": {
+        "message": "PID pitch",
```
Comment (mikeller) [inline locales/en/messages.json]: 'PID: Pitch'

File: locales/en/messages.json [hunk 12]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementPIDPitch": {
         "message": "Pitch axis PID gains"
     },
+    "osdTextElementPIDYaw": {
+        "message": "PID yaw",
```
Comment (mikeller) [inline locales/en/messages.json]: 'PID: Yaw'

File: locales/en/messages.json [hunk 13]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementPIDYaw": {
         "message": "Yaw axis PID gains"
     },
+    "osdTextElementPower": {
+        "message": "Power",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery: Power'

File: locales/en/messages.json [hunk 14]
```diff
@@ -3766,186 +3766,454 @@
...
     },
+    "osdTextElementBatteryWarning": {
+        "message": "Battery warning",
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementBatteryWarning": {
         "message": "Warning text that appears when the battery voltage falls below warning threshold"
     },
+    "osdTextElementAvgCellVoltage": {
+        "message": "Average cell voltage",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery average cell voltage'

File: locales/en/messages.json [hunk 15]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementAvgCellVoltage": {
         "message": "Average cell voltage (main battery voltage / cell count)"
     },
+    "osdTextElementPitchAngle": {
+        "message": "Pitch angle",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Angle: Pitch'

File: locales/en/messages.json [hunk 16]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementPitchAngle": {
         "message": "Numerical pitch angle in degrees"
     },
+    "osdTextElementRollAngle": {
+        "message": "Roll angle",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Angle: Roll'

File: locales/en/messages.json [hunk 17]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementRollAngle": {
         "message": "Numerical roll angle in degrees"
     },
+    "osdTextElementMainBattUsage": {
+        "message": "Main battery usage",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Battery usage;

File: locales/en/messages.json [hunk 18]
```diff
@@ -3766,186 +3766,454 @@
...
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementMainBattUsage": {
         "message": "Graphical representation of battery capacity usage"
     },
+    "osdTextElementArmedTime": {
+        "message": "Armed time",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Timer: Armed time'

File: locales/en/messages.json [hunk 19]
```diff
@@ -3766,186 +3766,454 @@
...
+        "message": "ESC RPM",
+        "description": "One of the elements of the OSD"
+    },
     "osdDescElementEscRpm": {
         "message": "RPM reported by ESC telemetry"
     },
-    "osdDescElementEscRpmFreq": {
-        "message": "RPM frequency reported by ESC telemetry"
+    "osdTextElementRemaningTimeEstimate": {
+        "message": "Remaining time estimate",
```
Comment (mikeller) [inline locales/en/messages.json]: 'Timer: Remaining time estimate'

**Comment by McGiverGim** [comment]
Thanks @mikeller!! This is the final result. I have changed your suggestions and some more:
image
image

**Comment by mikeller** [comment]
Looks mostly good - I wonder if we should add a special case to put all of the 'Unknown <x>' elements at the very end?

**Comment by McGiverGim** [comment]
I thought at it but I wasn't sure neither. I can do it as you prefer.

**Comment by mikeller** [comment]
I think it makes sense - these are a special case.

**Comment by McGiverGim** [comment]
Added a commit that moves the unknown fields to the end of the list

**Pull request merged**